File: pdf_merger.rb
#!/usr/bin/env ruby

require 'combine_pdf'

if ARGV.length < 2
  puts 'Usage: ruby pdf_merger.rb output.pdf input1.pdf input2.pdf ...'
  exit
end

output_file = ARGV[0]
input_files = ARGV[1..-1]

pdf = CombinePDF.new

input_files.each do |file|
  begin
    pdf << CombinePDF.load(file)
  rescue CombinePDF::Error
    puts "Error loading file: #{file}"
  end
end

pdf.save(output_file)
puts "PDF files merged successfully! Output file: #{output_file}"

File: pdf_merger.py
#!/usr/bin/env python3

import pypdf


def merge_pdfs(output_file, input_files):
    pdf_merger = pypdf.PdfMerger()

    for input_file in input_files:
        try:
            pdf_merger.append(input_file)
        except pypdf.utils.PdfReadError:
            print(f"Error loading file: {input_file}")

    with open(output_file, "wb") as output_pdf:
        pdf_merger.write(output_pdf)

    print(f"PDF files merged successfully! Output file: {output_file}")


if __name__ == "__main__":
    import argparse

    parser = argparse.ArgumentParser(description="Merges PDF files.")
    parser.add_argument("output_file", help="Output PDF file")
    parser.add_argument("input_files", nargs="+", help="Input PDF files")
    args = parser.parse_args()

    merge_pdfs(args.output_file, args.input_files)
